tools/release/common.sh

```diff
@@ -44,9 +44,9 @@ find_prev_release() {
   find_last_release
 
   if [[ $LAST_RELEASE_VERSION =~ "-rc" ]]; then
-    PREV_RELEASE_TAG=$(git tag --list --sort=taggerdate 'mimir-[0-9]*' | tail -2 | head -1)
```

Comment:
If we remove `--sort=taggerdate` the tags will be sorted lexicographically. I don't think it's correct, because it would break once we'll hit Mimir 2.10 in a couple of releases.

What we want here is getting the tags sorted by version. We could use `sort --version-sort` for that (e.g. `git tag --list 'mimir-[0-9]*' | sort --version-sort | tail -2 | head -1`) but the problem is that the release candidates wouldn't be sorted correctly.

For example, `git tag --list 'mimir-[0-9]*' | sort --version-sort` returns:

```
mimir-2.0.0
mimir-2.0.0-rc.0
mimir-2.0.0-rc.1
mimir-2.0.0-rc.2
mimir-2.0.0-rc.3
mimir-2.0.0-rc.4
mimir-2.1.0
mimir-2.1.0-rc.0
mimir-2.1.0-rc.1
mimir-2.2.0
mimir-2.2.0-rc.0
mimir-2.2.0-rc.1
mimir-2.3.0
mimir-2.3.0-rc0
mimir-2.3.0-rc.1
mimir-2.3.0-rc.2
mimir-2.3.1
mimir-2.4.0
mimir-2.4.0-rc.0
mimir-2.4.0-rc.1
mimir-2.5.0
mimir-2.5.0-rc.0
mimir-2.6.0
mimir-2.6.0-rc.0
mimir-2.6.1
mimir-2.7.0
mimir-2.7.1
mimir-2.7.2
mimir-2.8.0-rc.0
mimir-2.8.0-rc.1
mimir-2.8.0-rc.2
```

While what we want is:
```
mimir-2.0.0-rc.0
mimir-2.0.0-rc.1
mimir-2.0.0-rc.2
mimir-2.0.0-rc.3
mimir-2.0.0-rc.4
mimir-2.0.0
...
```

I've spent a good 30m playing with `sort` features, with no luck.

The hack I've found is to add a suffix which sort as last to each version not including "rc":

```
git tag --list 'mimir-[0-9]*' | gsed '/rc/b; s/\(.*\)/\1-xxx/' | sort --version-sort | gsed 's/-xxx//'
```

The output is:

```
mimir-2.0.0-rc.0
mimir-2.0.0-rc.1
mimir-2.0.0-rc.2
mimir-2.0.0-rc.3
mimir-2.0.0-rc.4
mimir-2.0.0
mimir-2.1.0-rc.0
mimir-2.1.0-rc.1
mimir-2.1.0
mimir-2.2.0-rc.0
mimir-2.2.0-rc.1
mimir-2.2.0
mimir-2.3.0-rc0
mimir-2.3.0-rc.1
mimir-2.3.0-rc.2
mimir-2.3.0
mimir-2.3.1
mimir-2.4.0-rc.0
mimir-2.4.0-rc.1
mimir-2.4.0
mimir-2.5.0-rc.0
mimir-2.5.0
mimir-2.6.0-rc.0
mimir-2.6.0
mimir-2.6.1
mimir-2.7.0
mimir-2.7.1
mimir-2.7.2
mimir-2.8.0-rc.0
mimir-2.8.0-rc.1
mimir-2.8.0-rc.2
```

NOTE: I've used `gsed` because OSX `sed` wouldn't work. If you adopt this approach, please use `$SED` and define it as `SED=$(which gsed || which sed)` (we already do it in `tools/release/create-draft-release-notes.sh` so we could just move it to `common.sh`.

Reply:
The change is applied as suggested.